### PR TITLE
Initialize the user before transfer-ownership

### PR DESCRIPTION
**Pull request by jvillafanez**
## Description
Initialize the user's FS if he hasn't logged in yet before transferring files to him.

## Related Issue

https://github.com/owncloud/enterprise/issues/3819

## Motivation and Context
Transfer was failing silently. The command output didn't show any error but the transfer failed

## How Has This Been Tested?

2. Create 2 users, user1 and user2 (local users)
3. Upload some files to user1
4. Without having logged in once with user2, execute `occ files:transfer-ownership user1 user2`
5. Files are transferred correctly to the user2

Checked also with master-key encryption **but not user-key encryption**

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by jvillafanez** [comment]
@phil-davis could you check the failing test? It seems to be caused by https://github.com/owncloud/core/blob/master/tests/lib/UtilTest.php#L388 and https://github.com/owncloud/core/blob/master/tests/lib/UtilTest.php#L411

I think the `$config->deleteSystemValue('skeletondirectory');` isn't being called due to the expected exception being thrown in those tests. This causes the system value not being properly reset, which affects this PR's tests.

**Comment by phil-davis** [comment]
@jvillafanez I pushed a commit that should fix that. Let's see how CI goes!

I also put the fix into a separate PR #37041 because it would be good to fix that UtilTest straight away, rather than waiting until this PR gets finished.

I wonder how many other places in the unit tests have "cleanup" code in a test case after an expected exception? Cleanup code should all go in the `tearDown()` method.

**Comment by phil-davis** [comment]
@jvillafanez If you pull this branch locally, rebase and force push, it should get rid of that 2nd commit - the `UtilTest` fixes are now merged in `master`

**Comment by jvillafanez** [comment]
I'll add the changelog entry after the review.

I have my doubts about including and using the `--force-user-init`. The only use case I see for the option is that you want to verify the destination user has logged in at least once. However, I think that normally you just want to transfer to that user.
I mean, we can remove the option and initialize the user without user confirmation.

**Comment by micbar** [comment]
@carlos @pako81 Can you confirm?

**Comment by pako81** [comment]
Having the `--force-user-init` input option would be maybe more transparent for the admin, to give him/her the information that the destination user has not log in yet.

On the other hand, final admin's aim is to just transfer files so, yes, I do agree with @jvillafanez that we could probably get rid of it.

File: changelog/unreleased/37038 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Bugfix: Initialize the user before the transfer-ownership command if the user has never logged in
+
+Trying to transfer the ownership of files to a user who hadn't logged in was causing problems
+because the FS of such user wasn't initialized and it wasn't possible to move the files there.
+The command appeared to work, but the files weren't moved.
+
+Now such user has the FS initialized to the transfer can be completed normally.
```
Comment (phil-davis) [inline changelog/unreleased/37038]:
```suggestion
Now such user has the FS initialized so the transfer can be completed normally.
```

**Comment by micbar** [comment]
@jvillafanez Please update the title and the top post with the new approach.

Rest looks fine IMO

**Comment by jvillafanez** [comment]
Done

**Review by micbar — APPROVED** [review]

**Pull request merged**